src/imitation/rewards/reward_nets.py

```diff
@@ -181,6 +182,8 @@ def predict_processed(
             action: Actions of shape `(batch_size,) + action_shape`.
             next_state: Successor states of shape `(batch_size,) + state_shape`.
             done: End-of-episode (terminal state) indicator of shape `(batch_size,)`.
+            kwargs: additional kwargs may be passed to change the functionality of
+                subclasses.
```

Comment:
To check I understand how this is meant to work -- I can just call `net.predict_processed(special_kwarg=42)` and have it either (a) used if `isinstance(net, MySpecialClass)`, and (b) silently ignored otherwise?

This has the downside that if I typo it to `spcial_kwarg=42)` then `MySpecialClass` will presumably also just ignore it? But it has the benefit I don't have to do extensive `isinstance` checks, and it's more ducktype-friendly as you can implement `EvenMoreSpecialClass` that also takes `special_kwarg`.

Reply:
Yes that is the tradeoff.

Reply:
However, there is a little more depth to it. The expected behavior is that all wrappers will pass along their `kwargs` to their base network. So in your example even if `net.base` was an instance of `MySpecialClass` it's `predict_processed` method should be called with special_kwarg=42. For an example of this see `test_normalization_wrappers_passes_on_kwargs`.

Reply:
Just realized that this was not the behavior of `ShapedRewardNet`. I've made the aforementioned test a bit more general, now it's called `test_wrappers_passes_on_kwargs`  and added a default implementation of `predict_processed` to `RewardNetWrapper`. This was all done in commit 23b1f4d0da4bff33efe8ae9e7eec53b6fd1873a4. So might want to look over the affected lines again.

Note: that I have not forced the add standard deviation wrapper to propagate `kwargs`. The reasons for this are: First, there are no current use cases for propagating `kwargs` through an ensemble. Second, it would require use to pass `kwargs`, intended for `predict_processed` to, `predict_reward_moments` which seems a bit odd. I could be convinced to change my mind on this, If you think there may be a valid use case for having the `kwargs` propagate though to `predict_reward_moments` or beyond to the individual ensemble members.

Reply:
I agree this smell's. Might be an instance of https://refactoring.guru/smells/message-chains. It also is related to #464 and #427. When we refactor reward nets we will need to accommodate for subclass specific configuration.